src/Cabal2JSON.hs

```diff
@@ -142,17 +178,26 @@ instance HasCodec PackageDescription where
         <*> requiredField' "extra-tmp-files" .= extraTmpFiles
         <*> requiredField' "extra-doc-files" .= extraDocFiles
 
+instance HasCodec (PerCompilerFlavor [String]) where
+  codec = dimapCodec (uncurry PerCompilerFlavor) (\(PerCompilerFlavor a b) -> (a, b)) codec
+
+instance (HasCodec a) => HasCodec (a, a) where
+  codec = dimapCodec (\(a:[b]) -> (a, b)) (\(a, b) -> [a, b]) codec
+
 instance HasCodec (CompilerFlavor, VersionRange) where
-  -- TODO: ?
   codec = undefined
```

Comment:
I am having some difficulties with this one. I cant transform the tuple into a list like I did in the lines above because the tuple contains 2 different types.

Reply:
@NorfairKing @marijanp

Reply:
@m1-s What do we want this to be? A string like `ghc >= 9.0`? An object `{ compiler: "ghc", versionRange: ">= 9.0"}`?
We have to figure that out first.
Then it's the same process to write a codec.
For example:

```
codec = object "(CompilerFlavor, VersionRange)" $ 
  (,) 
      <$> requiredField' "compiler" .= fst
      <*> requiredFiel'd "versionRange" .= snd
```

Reply:
The less we have to parse with nix the better. Therefore something like the second option `{ compiler: "ghc", versionRange: ">= 9.0"}` sounds better to me.